yarn.config.cjs

```diff
@@ -0,0 +1,50 @@
+/*
+% Enforces `exports` to be consistent
+gen_enforced_field(WorkspaceCwd, 'exports', '{ ".": "./lib/index.js", "./package.json": "./package.json" }') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true),
+  % Exclude packages with more complex `exports`
+  workspace_ident(WorkspaceCwd, WorkspaceIdent),
+  WorkspaceIdent \= '@babel/compat-data',
+  WorkspaceIdent \= '@babel/helper-plugin-test-runner', % TODO: Remove in Babel 8
+  WorkspaceIdent \= '@babel/core', % TODO: Remove in Babel 8
+  WorkspaceIdent \= '@babel/parser',
+  WorkspaceIdent \= '@babel/plugin-transform-react-jsx', % TODO: Remove in Babel 8
+  WorkspaceIdent \= '@babel/standalone',
+  WorkspaceIdent \= '@babel/types', % @babel/types has types exports
+  \+ atom_concat('@babel/eslint-', _, WorkspaceIdent),
+  \+ atom_concat('@babel/runtime', _, WorkspaceIdent).
+*/
+
+function enforcePackageExports({ Yarn }) {
+  for (const workspace of Yarn.workspaces()) {
+    if (workspace.manifest.private) continue;
+    // Exclude packages with more complex `exports`
+    const packageName = workspace.pkg.ident;
+    if (
+      [
+        "@babel/compat-data",
+        "@babel/helper-plugin-test-runner", // TODO: Remove in Babel 8
+        "@babel/core", // TODO: Remove in Babel 8
+        "@babel/parser",
+        "@babel/plugin-transform-react-jsx", // TODO: Remove in Babel 8
+        "@babel/standalone",
+        "@babel/types", // @babel/types has types exports
+      ].includes(packageName) ||
+      packageName.startsWith("@babel/eslint-") ||
+      packageName.startsWith("@babel/runtime")
+    ) {
+      continue;
+    }
+
+    workspace.set("exports", {
+      ".": "./lib/index.js",
+      "./package.json": "./package.json",
+    });
```

Comment:
@merceyz It seems that we can not force an object to be a package json field because the constraint validator now uses stringEqual. The prolog version works for Yarn 3 though.

https://github.com/babel/babel/actions/runs/6712269430/job/18241481440?pr=16069#step:4:7

I am aware that the prolog rules are deprecated and migrated the affected rule to js.

Reply:
This will be fixed in the next patch release (https://github.com/yarnpkg/berry/pull/5934)